Sources/VLCMediaList.m

```diff
@@ -163,17 +164,18 @@ - (void)insertMedia:(VLCMedia *)media atIndex: (NSUInteger)index
     libvlc_media_list_insert_media(p_mlist, [media libVLCMediaDescriptor], (int)index);
 }
 
-- (void)removeMediaAtIndex:(NSUInteger)index
+- (VLCMedia *)removeMediaAtIndex:(NSUInteger)index
 {
+    VLCMedia *target = [_mediaObjects objectAtIndex:index];
```

Comment:
`objectAtIndex` checks if the provided index is in range

Reply:
This will lead to an out-of-bounds exception and early application termination by the runtime if the index is not in range. Please add sanity checks.

Reply:
I personally think this is the desirable behavior for the end user. I had a hard time debugging things when `removeMediaAtIndex` function wasn't critically failing even when an invalid index was passed to it.